### ISSUE TITLE
IndexError in crypttab when /etc/crypttab is empty

### ISSUE THREAD

**Issue by kubilus1** [ansible/ansible-modules-extras#1979]
##### ISSUE TYPE

- Bug Report
##### COMPONENT NAME

crypttab.py
##### ANSIBLE VERSION

```
<!--- Paste verbatim output from “ansible --version” between quotes -->
```

2.1.0
##### OS / ENVIRONMENT

Red Hat Enterprise Linux Workstation release 6.6 (Santiago)
##### SUMMARY

The crypttab module contains an unchecked index into an array that can lead to an IndexError when the /etc/crypttab configuration file is emptied.

Currently, the crypttab module assumes that the /etc/crypttab file contains a trailing newline after the last entry.  If a user has edited /etc/crypttab and has not added the trailing newline, if the crypttab module removes the last crypptab entry the following error occurs:

failed: [cb] => {"failed": true, "parsed": false}
BECOME-SUCCESS-atlkxchqoylvcpkxtyiajfajozptnvwx
Traceback (most recent call last):
  File "/root/.ansible/tmp/ansible-tmp-1459953324.47-72446787071472/crypttab", line 1973, in <module>
    main()
  File "/root/.ansible/tmp/ansible-tmp-1459953324.47-72446787071472/crypttab", line 163, in main
    f.write(str(crypttab))
  File "/root/.ansible/tmp/ansible-tmp-1459953324.47-72446787071472/crypttab", line 208, in **str**
    if crypttab[-1] != '\n':
IndexError: string index out of range
OpenSSH_5.3p1, OpenSSL 1.0.1e-fips 11 Feb 2013
debug1: Reading configuration data /home/mkubilus/.ssh/config
debug1: Applying options for *
debug1: Reading configuration data /etc/ssh/ssh_config
debug1: Applying options for *
debug1: auto-mux: Trying existing master
debug1: mux_client_request_session: master session id: 2
debug1: mux_client_request_session: master session id: 2
Shared connection to 10.125.87.225 closed.

This is due to line 208 in crypttab.py not checking the length of the array before indexing into the array.
##### STEPS TO REPRODUCE

Set /etc/crypptab to contain a single line, for example:

```
foo /dev/sdb /keys/bar
```

Setup playbook to remove line:

```
crypptab: name=foo state=absent password=/keys/bar backing_dev=/dev/sdb
```

Run the playbook and receive IndexError.

##### EXPECTED RESULTS

Crypttab module does not through python exception.
##### ACTUAL RESULTS

Python exception occurs:
Traceback (most recent call last):
  File "/root/.ansible/tmp/ansible-tmp-1459953324.47-72446787071472/crypttab", line 1973, in <module>
    main()
  File "/root/.ansible/tmp/ansible-tmp-1459953324.47-72446787071472/crypttab", line 163, in main
    f.write(str(crypttab))
  File "/root/.ansible/tmp/ansible-tmp-1459953324.47-72446787071472/crypttab", line 208, in **str**
    if crypttab[-1] != '\n':
IndexError: string index out of range

```
<!--- Paste verbatim command output between quotes -->
```

**Comment by kubilus1** [comment]
Resolved by:  #1980

**Comment by mscherer** [comment]
Since 1980 was merged, I am closing this issue.
